### ISSUE TITLE
Query parameters are incorrectly sorted

### ISSUE THREAD

**Issue by CarmenPopoviciu** [bent0b0x/amazon-mws#1]
Consider the setup below:

```
mwsClient({
    method: 'POST',
    base: mwsBase,
    endpoint: mwsEndpoint,
    params: {
      'Action': 'ListOrders',
      'CreatedAfter': new Date(params.createdAfter).toISOString(),
      'OrderStatus.Status.1': 'Unshipped',
      'OrderStatus.Status.2': 'PartiallyShipped',
      'MarketplaceId.Id.1': mwsMarketplaceId,
      'SellerId': mwsSellerId,
      'Version': mwsVersion
    });
```

The API response for the request generated based on those parameters is:

```
<ErrorResponse xmlns="https://mws.amazonservices.com/Orders/2013-09-01">
  <Error>
    <Type>Sender</Type>
    <Code>SignatureDoesNotMatch</Code>
    <Message>The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.</Message>
  </Error>
  <RequestID>...</RequestID>
</ErrorResponse>
```

After looking into it and comparing the query string generated by the lib with the one returned by the amazon [scratchpad](https://mws.amazonservices.com/scratchpad/index.html), for the exact same request, I noticed that that the order of the query parameters was different. The sorted params array, returned by the library is:

```
[ [ 'SellerId', '...' ],
  [ 'Action', 'ListOrders' ],
  [ 'AWSAccessKeyId', '...' ],
  [ 'CreatedAfter', '...' ],
  [ 'MarketplaceId.Id.1', '...' ],
  [ 'OrderStatus.Status.1', 'Unshipped' ],
  [ 'OrderStatus.Status.2', 'PartiallyShipped' ],
  [ 'SignatureMethod', 'HmacSHA256' ],
  [ 'SignatureVersion', '2' ],
  [ 'Timestamp', '2016-07-28T12:23:51.683Z' ],
  [ 'Version', '2013-09-01' ] ]
```

As far as I can tell, the mws [docs](http://docs.developer.amazonservices.com/en_US/dev_guide/DG_ClientLibraries.html) state that the parameters should be alphabetically ordered, which is obviously not the case for the above. Looking at the code, I see the lib uses [this](https://github.com/bent0b0x/amazon-mws/blob/master/amazon.js#L23) as the sorting function for the params array. AFAIK, sort's `compareFunction` should return either a positive/negative number or 0, which, unless I'm missing something, the current function in that code, doesn't. 

I think this can easily be fixed by just using `.sort()` with the default behaviour of `compareFunction` and I would be more than happy to submit a PR for this asap

**Comment by joanniclaborde** [comment]
+1!

**Comment by danazkari** [comment]
Seems like this project has been sort of abandoned... do you guys know if this project is still relevant? I don't mind forking and applying the change #2 and using that instead of official repo, if that makes it work properly.

**Comment by bent0b0x** [comment]
Sorry, did not realize anyone had been using this. Will look into this, generally improving the repo, and merging outstanding PRs.

**Comment by bent0b0x** [comment]
Fixed by #3
